keyboards/oddball/adns/adns.c

```diff
@@ -1,9 +1,23 @@
-#include <avr/io.h>
-#include <avr/interrupt.h>
+/* Copyright 2020 Alexander Tulloh
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #include "quantum.h"
-#include "pointing_device.h"
-#include "adns9800_srom_A4.h"
+#include "adns9800_srom_A6.h"
 #include "../../lib/lufa/LUFA/Drivers/Peripheral/SPI.h"
+#include "adns.h"
```

Comment:
I think these can be replaced with spi.h

Reply:
Yep, this should be using spi_master: https://docs.qmk.fm/#/spi_driver

Reply:
Hi, I've now done this, and is working as expected. Just some things I came across:
- I needed to add `spi_master.h` manually to my calling code
- I had to add `src += spi_master.c` to my `rules.mk` to get it to compile, I couldn't find a particular build option anywhere
- I'm not really too familiar with SPI, but I had to `setPinOutput(SPI_SS_PIN)` before calling `spi_init()` to get things working (I saw it did the same in the LUFA SPI_AVR8.h `SPI_Init(...)` function)

Reply:
Yeah, those would all be the correct way to handle things. 

As for the last bit. I think that's a bug, actually. And specifically, somebody called the wrong function:
https://github.com/qmk/qmk_firmware/blob/0f9b7b9768e53add15dc7e860efab0464cd7533d/drivers/avr/spi_master.c#L45-L52